[CI] `linux://python/ray/serve:test_standalone2` is failing/flaky on master.

- e043f49957a16c706d269fa0658c314fea8a6e5a FAILED [Buildkite :serverless: Serve Tests](https://buildkite.com/ray-project/ray-builders-branch/builds/8781#01821e00-c923-48df-9d68-3a86f0756074)
- 8271406a04e380421c6226cbec1b6ea55189ea72 FAILED [Buildkite :serverless: Serve Tests](https://buildkite.com/ray-project/ray-builders-branch/builds/8779#01821dd5-8038-4c66-bd12-372707c90380)
- 7f908c408666acc24ca68ca3e45fa73755c1dd4f FAILED [Buildkite :serverless: Serve Tests](https://buildkite.com/ray-project/ray-builders-branch/builds/8778#01821d95-7241-4fed-b218-f8a4895b7b96)
- e7957f4a3ee92cd0869b453091b253e3217b3eb3 FAILED [Buildkite :serverless: Serve Tests](https://buildkite.com/ray-project/ray-builders-branch/builds/8775#01821d8d-6953-4553-9b58-ec9d06329847)
- 6a71a293b2aef733a4047fd580640e88aad03545 FAILED [Buildkite :serverless: Serve Tests](https://buildkite.com/ray-project/ray-builders-branch/builds/8764#01821cd9-ece9-4a29-ab2d-b6526abdeca4)
- e157989a7a0f222d38b0915b0dba1d3652031422 FAILED [Buildkite :serverless: Serve Tests](https://buildkite.com/ray-project/ray-builders-branch/builds/8761#01821a3a-157c-4e74-969e-421b995b47b3)
- 887f6726d552be38ab79811dd070af1247189ec8 FAILED [Buildkite :serverless: Serve Tests](https://buildkite.com/ray-project/ray-builders-branch/builds/8757#018219b4-4b42-4101-b6e7-474adafcd7fc)

....
Generated from flaky test tracker. Please do not edit the signature in this section.
DataCaseName-linux://python/ray/serve:test_standalone2-END
....

can repro locally for 
```
FAILED test_standalone2.py::test_controller_recover_and_delete - AssertionError: assert 2 > 2
```

however it is definitely flaky

--- a/python/ray/serve/tests/test_standalone2.py
+++ b/python/ray/serve/tests/test_standalone2.py
@@ -585,7 +585,7 @@ def test_controller_recover_and_delete(shutdown_ray):
 
     actors = ray.util.list_named_actors(all_namespaces=True)
 
-    # Try to delete the deployments and kill the controlle rright after
+    # Try to delete the deployments and kill the controller right after
     client.delete_deployments(["f"], blocking=False)
     ray.kill(client._controller, no_restart=False)
 

--- a/python/ray/serve/tests/test_standalone2.py
+++ b/python/ray/serve/tests/test_standalone2.py
@@ -36,16 +36,10 @@ def shutdown_ray():
 def start_and_shutdown_ray_cli():
     subprocess.check_output(
         ["ray", "start", "--head"],
-        stdout=subprocess.STDOUT,
-        stderr=subprocess.STDOUT,
-        shell=True,
     )
     yield
     subprocess.check_output(
         ["ray", "stop", "--force"],
-        stdout=subprocess.STDOUT,
-        stderr=subprocess.STDOUT,
-        shell=True,
     )
 
 

--- a/python/ray/serve/tests/test_standalone2.py
+++ b/python/ray/serve/tests/test_standalone2.py
@@ -34,9 +34,19 @@ def shutdown_ray():
 
 @contextmanager
 def start_and_shutdown_ray_cli():
-    subprocess.check_output(["ray", "start", "--head"])
+    subprocess.check_output(
+        ["ray", "start", "--head"],
+        stdout=subprocess.STDOUT,
+        stderr=subprocess.STDOUT,
+        shell=True,
+    )
     yield
-    subprocess.check_output(["ray", "stop", "--force"])
+    subprocess.check_output(
+        ["ray", "stop", "--force"],
+        stdout=subprocess.STDOUT,
+        stderr=subprocess.STDOUT,
+        shell=True,
+    )
 
 
 @pytest.fixture(scope="function")
@@ -558,7 +568,7 @@ class TestDeployApp:
         assert client.get_serve_status().app_status.deployment_timestamp == 0
 
 
-def test_controller_recover_and_delete():
+def test_controller_recover_and_delete(shutdown_ray):
     """Ensure that in-progress deletion can finish even after controller dies."""
 
     ray.init()
@@ -574,17 +584,16 @@ def test_controller_recover_and_delete():
     f.deploy()
 
     actors = ray.util.list_named_actors(all_namespaces=True)
-    client.delete_deployments(["f"], blocking=False)
 
+    # Try to delete the deployments and kill the controlle rright after
+    client.delete_deployments(["f"], blocking=False)
+    ray.kill(client._controller, no_restart=False)
+
+    # All replicas should be removed already or after the controller revives
     wait_for_condition(
         lambda: len(ray.util.list_named_actors(all_namespaces=True)) < len(actors)
     )
-    ray.kill(client._controller, no_restart=False)
 
-    # There should still be replicas remaining
-    assert len(ray.util.list_named_actors(all_namespaces=True)) > 2
-
-    # All replicas should be removed once the controller revives
     wait_for_condition(
         lambda: len(ray.util.list_named_actors(all_namespaces=True)) == len(actors) - 50
     )

